oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/requests/search/aggregation/TermsAggregation.java

```diff
@@ -58,6 +58,8 @@ public void serialize(final TermsAggregation value, final JsonGenerator gen,
                     if (value.getOrder() != null) {
                         writeOrder(value, gen);
                     }
+                    gen.writeStringField("execution_hint", "map");
+                    gen.writeStringField("collect_mode", "breadth_first");
```

Comment:
You put these changes in the `TermsAggregation`, I think this would affect not just topology, it could affect `TopN` query, such as slow instance latency list.

I could find the doc of ElasticSearch, https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-terms-aggregation.html#search-aggregations-bucket-terms-aggregation-collect

So, are you meaning to do level-one aggregation first always provides better performance?

Reply:
please see here https://www.elastic.co/guide/en/elasticsearch/reference/6.8/search-aggregations-bucket-terms-aggregation.html#CO114-1


Because the metrics data is relatively small, using breadth_first, and execution_hint map will have a huge query improvement。

I did a simple test
When the index data is empty, "execution_hint" is not added, "map" took 30209ms, and the query timeout

When adding "execution_hint ""map" took 556ms

There are also similar question by other https://stackoverflow.com/questions/50893653/elasticsearch-slow-terms-aggregation/50895792#50895792

Reply:
> 

this elasticsearch version 6.8.0

Reply:
Make sense. 2 things

1. Please update `changes.md` in the doc, we have ref in the PR template(but you deleted it)
2. Moving this change into ElasticSearch DAO. `TermsAggregation` is our version of the ElasticSearch client, it should not include OAP's internal logic.

Others are good to me.